### DISCUSSION TITLE
Measuring encoding and decoding time of format images

### DISCUSSION THREAD

**Discussion by Ciepielewski-Cezar** [dlemstra/Magick.NET#910] [Help] [answered]
Hello
I want measure encoding and decoding time for few image formats. From what I understand, encoding takes place, if we take an image in any format and convert it to the target format, e.g. from .png to .jpg. Decoding takes place, when we open any image e.g. in a photo viewing program.

So, if I want measure encoding time, I need to read the image, convert to target format and measure time for the convert operation. For decoding time, I need only measure time for reading the image, in target format, as a `MagickImage`. Here is my code for measure times for jpeg format:

```c#
var defaultImage = new MagickImage("i1.png");

var encodingTime = Stopwatch.StartNew();
defaultImage.Write($"i1.jpg");
encodingTime.Stop();

var decodingTime = Stopwatch.StartNew();
var targetImage = new MagickImage("i1.jpg");
decodingTime.Stop();
```
However i am not sure that, my thinking is correct, and I have some doubts. 
When a  `MagickImage` object is converting from one format to another by `Write` function, only the encoding operation takes place, or both, decoding from a first format, and encoding to a second?
When I am reading a image as  `MagickImage` object, decoding takes place, or do I need use something another?

I'm sorry if these questions seem easy, I just want to know, if I'm doing something wrong.

**Comment by dlemstra** [comment, marked as answer]
Those are indeed the correct methods to measure encoding and decoding.
